Stratis.SmartContracts/UInt256.cs

```diff
@@ -157,39 +157,44 @@ public UInt256(byte[] vch) : this(vch, true)
             return new UInt256(value);
         }
 
-        public static implicit operator UInt256(BigInteger value)
-        {
-            return new UInt256(value);
-        }
-
         public static implicit operator UInt256(UInt128 value)
         {
-            return new UInt256(value);
+            var bytes = new byte[32];
+            value.ToBytes().CopyTo(bytes, 0);
+            return new UInt256(bytes);
         }
 
-        public static implicit operator int(UInt256 value)
+        public static explicit operator int(UInt256 value)
         {
             return (int)value.value.GetValue();
         }
 
-        public static implicit operator uint(UInt256 value)
+        public static explicit operator uint(UInt256 value)
         {
             return (uint)value.value.GetValue();
         }
 
-        public static implicit operator long(UInt256 value)
+        public static explicit operator long(UInt256 value)
         {
             return (long)value.value.GetValue();
         }
 
-        public static implicit operator ulong(UInt256 value)
+        public static explicit operator ulong(UInt256 value)
         {
             return (ulong)value.value.GetValue();
         }
 
-        public static implicit operator BigInteger(UInt256 value)
+        public static explicit operator UInt128(UInt256 value)
         {
-            return value.value.GetValue();
```

Comment:
Can this stay as `new UInt128(value.value.GetValue())` and make private constructor as internal ?

Reply:
See comment below.